data/src/main/java/com/globant/equattrocchio/data/ImagesServicesImpl.java

```diff
@@ -13,28 +16,32 @@
 
 public class ImagesServicesImpl implements ImagesServices {
 
-    private static final String URL= "http://splashbase.co/";
+    private static final String URL = "http://splashbase.co/";
 
     @Override
-    public void getLatestImages(Observer<Boolean> observer) {
+    public void getLatestImages(@NonNull final Observer<ImageList> observer) {
         Retrofit retrofit = new Retrofit.Builder().
                 baseUrl(URL).
                 addConverterFactory(GsonConverterFactory.create())
                 .build();
 
-        SplashbaseApi api  = retrofit.create(SplashbaseApi.class);
+        SplashbaseApi api = retrofit.create(SplashbaseApi.class);
 
         Call<Result> call = api.getImages();
 
         call.enqueue(new Callback<Result>() {
             @Override
             public void onResponse(Call<Result> call, Response<Result> response) {
                 //todo: show the response.body() on the ui
+
+                observer.onNext(ResultToImageListMapper.map(response.body()));
             }
 
             @Override
             public void onFailure(Call<Result> call, Throwable t) {
                 //todo: update the UI with a connection error message
+
+                observer.onNext(null);
```

Comment:
If you sent a null in a onNext, Rx crashes.

Reply:
Done